### ISSUE TITLE
Manually invoking resolvers through code

### ISSUE THREAD

**Issue by nikolakanacki** [graphql-compose/graphql-compose#114]
For example call `PostTC.get('$findById').call([context, args, fields, ...stuff])`. Is there a public/documented way of doing this?

**Comment by nodkz** [comment]
Yep, you may call it in such way `PostTC.getResolver('findById').resolve(rp)`.

Where `rp` is an object of ResolveParams with the following shape:
 ```js
export type ResolveParams<TSource, TContext> = {
  source: TSource,
  args: { [argName: string]: any },
  context: TContext,
  info: GraphQLResolveInfo,
  projection: $Shape<ProjectionType>,
  [opt: string]: any,
};
```

So for finding post by id you may execute following command:
```js
PostTC.getResolver('findById').resolve({ args: { _id: '5aab82d5bc94830001791f08' }});

// or
PostTC.getResolver('findOne').resolve({ args: { filter: { _id: '5aab82d5bc94830001791f08' }}});
```

**Comment by nodkz** [comment]
PS. Following code almost identical
- `PostTC.get('$findById')` - magic method for getting data by path `$` - resolver, `@` - argument, 'without prefix' - regular field
- `PostTC.getResolver('findById')` - recommended way for getting resolver, it's staticaly typed and if you use Flowtype/Typescript it will provide autosuggestion and static analysis checks

**Comment by nikolakanacki** [comment]
@nodkz Thanks, makes things a lot more clearer! Any way to call subfield resolvers without a `graphql` query or do a `graphql` "selection" on fields without it?

**Comment by nodkz** [comment]
Sorry, no way. Calling subfields resolvers is graphql care. Otherwise graphql-compose should duplicate same logic in resolvers.

Anyway it's a good feature request. And maybe will be found easy way how to do it in the future.

**Comment by nikolakanacki** [comment]
@nodkz It may be done by passing a "local" graphql statement to the resolver? Something like:
```js
PostTC.get('$findById').resolve({
  args: {},
  graphql: {
    variables: {},
    // This one could be wrapped before parsing to remove boilerplate like so:
    query: `
      field,
      anotherField(arg1: $arg1) {
        ...
      }
    `,
  },
});

**Comment by nodkz** [comment]
Nope. I don't want to add additional logic to `resolve` method for serving `graphql` arg. Cause it adds performance degradation and makes resolver logic dirtier.

So I recommend creating your own helper method, something like this:
```js
myResolverExecutor({
  resolver: PostTC.get('$findById'),
  query: ``,
  variables: {},
})
```
It should from query&variables prepare `info` arg for resolver, execute it and return payload.

**Comment by rockus123** [comment]
For something simple apis, graphql and libraries still have no standard, making queries and relations hard when calling things directly.  A lot of time, the use case is get me this with some filter, and any child relationship with filter and get only these elements we want.  Graphql tries to solve but only one part.

**Comment by aks30498** [comment]
> Maybe we need to create a separate function `Resolver.exec` which will execute resolver in GraphQL manner. Something like:
> 
> ```js
> PostTC.get('$findById').exec({ // or `run` don't know which name will be better
>   query: `
>       field,
>       anotherField(arg1: $arg1) {
>         ...
>       }
>   `,
>   variables: {}, // optional, but required if used args in query
>   source: {}, // optional, but required if resolve method use source arg
>   context: {}, // optional, but required if resolve method use context
> });
> ```
> 
> Under the hood, it must prepare `source`, `args`, `context` and `info` params and call standard `resolve` method.
> 
> I think this method will be useful for testing and custom calls, but it quite complex for realization. Need more votes for implementation of this feature.

Do we have an exec function now? I'm trying to manually resolve like so 
```
DayTC.mongooseResolvers
                    .dataLoader()
                    .resolve({ args: { _id } })
```

But it says `Cannot use Day.dataLoader resolver without 'info: GraphQLResolveInfo'` 

I am new and I have no idea where to get that info param to be passed to it :/ 

here is the whole thing: 
```  dayRemoveById: DayTC.mongooseResolvers
        .removeById()
        .wrapResolve((next) => async (rp) => {
            try {
                const {
                    args: { _id },
                } = rp;

                let day = await DayTC.mongooseResolvers
                    .dataLoader()
                    .resolve({ args: { _id } });

                if (day.trackedTasks.length > 0) {
                    console.log(day);

                    await Task.updateMany(
                        {
                            _id: { $in: trackedTasks.map((tt) => tt.task) },
                        },
                        { $pull: { records: { day: _id } } }
                    );

                    return next(rp);
                }
            } catch (error) {
                console.log(error);
                return null;
            }
        }),```
        
Basically I want to remove records from another collection related to the day if that day is removed.

**Comment by nodkz** [comment]
@aks30498 in your case better to fetch Day via mongoose model directly without calling `resolver.resolve`.

-----

PS. For future googlers:
If you want to call `resolver.resolve` method you must pass all required params. It depends on the internal implementation of the resolver. From example above, `TC.mongooseResolvers.dataLoader()` requires `context` & `info` params and you may pass them from parent resolver:
```js
 dayRemoveById: DayTC.mongooseResolvers
        .removeById()
        .wrapResolve((next) => async (rp) => {
            let day = await DayTC.mongooseResolvers
                    .dataLoader()
                    .resolve({ args: { _id }, context: rp.context, info: rp.info });
        })
```

Anyway better not to do this. I recommend writing your custom resolver from scratch without complex resolver wrapping which is difficult in maintaining.
